### PR TITLE
Misc Cleanups

### PR DESCRIPTION
**Pull request by Myoldmopar**
Mostly just changes to bring the testing frameworks and release packaging up to speed.  Since we'll be trying to integrate this with the E+ core installer, I wanted to convert the Travis and AppVeyor builds over to Github Actions, and get it running on the same build configurations as we built E+ itself.  This is basically done though will need some tidying.  

The Linux packages are _huge_ so I need to see what that is including.  The next PR will address that along with starting to make a pass over the codebase itself to clean up any obvious issues.  Also need to get the workflow tester built and packaged on each platform.

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -8,7 +8,7 @@ omit =
         .tox/*
         *.pyenv*
         *.venv*
-
+        */AppData/*
```
Comment (Myoldmopar) [inline .coveragerc:11]: On Github Action WIndows instances, the Python install lives in `AppData`, so we need to ignore that directory when calculating the coverage results.

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+ignore = E501
+max-line-length = 100
+exclude = .git,.tox,conf.py,setup.py
```
Comment (Myoldmopar) [inline .flake8:4]: This was pulled from the `tox.ini` file as this is more appropriate anyway.

File: .github/workflows/flake8.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+
+    - name: Install Pip Dependencies
+      run: pip install flake8
+
+    - name: Run Flake8
+      run: flake8 eplaunch
```
Comment (Myoldmopar) [inline .github/workflows/flake8.yml:21]: A new Github Workflow for running Flake8 on the codebase.

File: .github/workflows/releases.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    - name: Build
+      shell: bash
+      run: pyinstaller ${{ matrix.pyinstall_args }} eplaunch.spec
+        # pyinstaller wf_tester.spec
+
+    - name: Upload
+      uses: actions/upload-artifact@v2
+      with:
+        name: EPLaunch-${{ matrix.os }}
+        path: dist/EPLaunch
```
Comment (Myoldmopar) [inline .github/workflows/releases.yml:46]: A new Github Workflow for building packages on all the platforms.

File: .github/workflows/unit_tests.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    - name: Install Pip Dependencies
+      shell: bash
+      run: pip install ${{ matrix.pip_arg }} coveralls nose ${{ matrix.wxpython }}
+    - name: Run Tests
+      shell: bash
+      run: ${{ matrix.python }} setup.py nosetests
+    - name: Upload Coverage Results
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: coveralls
```
Comment (Myoldmopar) [inline .github/workflows/unit_tests.yml:44]: A new Github Workflow for running unit tests and posting results to Coveralls.

File: .gitignore [hunk 6]
```diff
@@ -2,6 +2,7 @@
 .idea
 tmp_build
 EPLaunchInstall.exe
+cover/
```
Comment (Myoldmopar) [inline .gitignore:5]: The coverage results will be in a `cover` directory, so have Git ignore that.

File: .noserc [hunk 7]
```diff
@@ -0,0 +1,4 @@
+[nosetests]
+with-coverage=1
+cover-erase=1
+cover-html=1
```
Comment (Myoldmopar) [inline .noserc:4]: Moved `nose` configuration to a specific nose config file.

File: README.md [hunk 8]
```diff
@@ -1,31 +1,34 @@
 # EP-Launch3
 
-[![GitHub release](https://img.shields.io/github/release/nrel/ep-launch.svg)](https://github.com/nrel/ep-launch/releases/latest)
+[![GitHub release](https://img.shields.io/github/release/nrel/ep-launch.svg?style=for-the-badge)](https://github.com/nrel/ep-launch/releases/latest)
```
Comment (Myoldmopar) [inline README.md:3]: Updated badge styles in the readme, and moved links from old test results to Github Actions.  Also updated commentary as needed.

File: eplaunch/interface/filenamemenus.py [hunk 9]
```diff
@@ -36,8 +36,8 @@ def get_file_list(self):
             if menu_item.GetId() == self.end_separator_id:
                 mode = False
             if mode:
-                list_of_menu_item_labels.append(menu_item.GetLabel())
-                self.file_name_to_id[menu_item.GetLabel()] = menu_item.GetId()
+                list_of_menu_item_labels.append(menu_item.GetItemLabel())
+                self.file_name_to_id[menu_item.GetItemLabel()] = menu_item.GetId()
```
Comment (Myoldmopar) [inline eplaunch/interface/filenamemenus.py:40]: My IDE was complaining about `GetLabel` -- need to test this on all the platforms prior to merging.

File: requirements.txt [hunk 10]
```diff
@@ -49,9 +20,13 @@
...
+
+# for running tests, you'll need the nose and coverage packages
+coverage==5.3
 nose==1.3.7
-pyinstaller==3.3.1
-pypubsub==4.0.0
-tox==2.9.1
+
+# for packaging up EP-Launch, you'll need pyinstaller:
+pyinstaller==4.1
```
Comment (Myoldmopar) [inline requirements.txt:32]: Updated with more info, but maybe needs even more updates with further testing.

**Review by Myoldmopar — COMMENTED** [review]
This needs to be tested on all the platforms, but if it seems to perform well I think it's ready to merge.

**Comment by Myoldmopar** [comment]
The packages are built by Github Actions now, and you can find them, for now, as artifacts on the action build: https://github.com/NREL/EP-Launch/runs/1480700521?check_suite_focus=true

File: eplaunch/interface/filenamemenus.py (hunk 9 above)
Comment (JasonGlazer) [reply eplaunch/interface/filenamemenus.py:40]: Works on Windows 10

**Review by JasonGlazer — APPROVED** [review]
Looks good.

**Comment by JasonGlazer** [comment]
@Myoldmopar it looks like coveralls did not work quite right when I merged this:

https://github.com/NREL/EP-Launch/runs/2521046561?check_suite_focus=true

**Pull request merged**